addons/account/models/account_move_line.py

```diff
@@ -2265,168 +2669,8 @@ def _add_exchange_difference_cash_basis_vals(self, exchange_diff_vals):
-                )
-
-                # Exchange difference for cash basis entries.
-                if is_cash_basis_needed:
-                    caba_lines_to_reconcile = involved_lines._add_exchange_difference_cash_basis_vals(exchange_diff_vals)
-
-                # Create the exchange difference.
-                if exchange_diff_vals['move_vals']['line_ids']:
-                    exchange_move = involved_lines._create_exchange_difference_move(exchange_diff_vals)
-                    if exchange_move:
-                        exchange_move_lines = exchange_move.line_ids.filtered(lambda line: line.account_id == account)
-
-                        # Track newly created lines.
-                        involved_lines += exchange_move_lines
-
-                        # Track newly created partials.
-                        exchange_diff_partials = exchange_move_lines.matched_debit_ids \
-                                                 + exchange_move_lines.matched_credit_ids
-                        involved_partials += exchange_diff_partials
-                        results['exchange_partials'] += exchange_diff_partials
-
-            # ==== Create the full reconcile ====
-            results['full_reconcile'] = self.env['account.full.reconcile'] \
-                .with_context(
-                    skip_invoice_sync=True,
-                    skip_invoice_line_sync=True,
-                    skip_account_move_synchronization=True,
-                    check_move_validity=False,
-                ) \
-                .create({
-                    'exchange_move_id': exchange_move and exchange_move.id,
-                    'partial_reconcile_ids': [Command.set(involved_partials.ids)],
-                    'reconciled_line_ids': [Command.set(involved_lines.ids)],
-                })
-
-            # === Cash basis rounding autoreconciliation ===
-            # In case a cash basis rounding difference line got created for the transition account, we reconcile it with the corresponding lines
-            # on the cash basis moves (so that it reaches full reconciliation and creates an exchange difference entry for this account as well)
-
-            if caba_lines_to_reconcile:
-                for (dummy, account, repartition_line), amls_to_reconcile in caba_lines_to_reconcile.items():
-                    if not account.reconcile:
-                        continue
-
-                    exchange_line = exchange_move.line_ids.filtered(
-                        lambda l: l.account_id == account and l.tax_repartition_line_id == repartition_line
-                    )
-
-                    (exchange_line + amls_to_reconcile).filtered(lambda l: not l.reconciled).reconcile()
-
-        not_paid_invoices.filtered(lambda move:
-            move.payment_state in ('paid', 'in_payment')
-        )._invoice_paid_hook()
-
-        return results
+        """ Reconcile the current move lines all together. """
+        return self._reconcile_plan([self])
```

Comment:
As written in the other PR, I think this should be the standard API to call in order to hide `_reconcile_plan` from the rest.

Every client module should call `reconcile` either amls or a list of batch of amls to reconcile.

Reply:
Just for the records: blablabla we had an IRL discussion about `reconcile` that could be called either as an `@api.model` using only the parameters or, when parameter aren't used, a normal method using only `self`.

Anyway, for the future, in order to have a clean design and a smart API that is convenient to use, it would be nice to have a pipe that only process plans and that we just transform amls into a plan to process.